internal/rest/avi_obj_vsvip.go

```diff
@@ -119,7 +119,12 @@ func (rest *RestOperations) AviVsVipBuild(vsvip_meta *nodes.AviVSVIPNode, vsCach
 						if vsvip.Vip[0].IPAMNetworkSubnet == nil {
 							vip.IPAMNetworkSubnet = &avimodels.IPNetworkSubnet{}
 						} else {
-							vip.IPAMNetworkSubnet = vsvip.Vip[0].IPAMNetworkSubnet
+							ipamNetworkRefSplit := strings.Split(*vsvip.Vip[0].IPAMNetworkSubnet.NetworkRef, "#")
+							if len(ipamNetworkRefSplit) > 1 && ipamNetworkRefSplit[1] != vsvip_meta.VipNetworks[0].NetworkName {
+								vip.IPAMNetworkSubnet = &avimodels.IPNetworkSubnet{}
+							} else {
+								vip.IPAMNetworkSubnet = vsvip.Vip[0].IPAMNetworkSubnet
+							}
 						}
 						vip.DiscoveredNetworks = vsvip.Vip[0].DiscoveredNetworks
```

Comment:
@DixitAakash: This change is irrelevant to your change. can you take out line number 129? It will be fixed from controller end.